bot/plotting/comparison_generator.py

```diff
@@ -75,42 +75,37 @@ def comparison_generator(
         diff_params = random.randint(2, 3)
         min_param_value = 100
         for i in range(0, diff_params):
-            # copy the original values and append them in the list
-            param_list.append(params.copy())
 
-            # generate a random value
+            # generate parameter values
             if (
                 param_to_vary == "Electrode height [m]"
                 or param_to_vary == "Electrode width [m]"
             ):
-                param_value = parameter_value_generator(
+                params, varied_value = parameter_value_generator(
                     chemistry, param_to_vary, lower_bound=0
                 )
             elif param_to_vary == "Ambient temperature [K]":
```

Comment:
This is ok for now, but if we end up needing to have more bounds you could make `param_to_vary_list` a nested dict instead with the bounds, e.g. `{"Electrode height [m]": {"lower bound": 0, "upper bound": None}}`